tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -337,18 +367,16 @@ private void AsmDataProductConfigChanged(object sender, ProductConfigChangedEven
             {
                 if (asmDataConfig.TypedFile?.RulesData?.Length > 0)
                 {
-                    _remoteConfigurationStatus.RulesData.AddRange(asmDataConfig.TypedFile.RulesData);
+                    _remoteConfigurationStatus.RulesDataByFile[asmDataConfig.Name] = asmDataConfig.TypedFile.RulesData.ToList();
```

Comment:
Do we need this to be a list? Could we keep the array?

Reply:
fixed in 995730f4b5c43d00fc92cf0d4231e2b7c761f62f